src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -356,11 +356,18 @@ public BitbucketCommit call() throws Exception {
     }
 
     private void setupClosureForPRBranch(BitbucketServerPullRequest pr) {
-        BitbucketServerBranch branch = (BitbucketServerBranch) pr.getSource().getBranch();
-        branch.setCommitClosure(new CommitClosure(branch.getRawNode()));
-
-        branch = (BitbucketServerBranch) pr.getDestination().getBranch();
-        branch.setCommitClosure(new CommitClosure(branch.getRawNode()));
+        try {
+            BitbucketServerBranch branch = (BitbucketServerBranch) pr.getSource().getBranch();
+            if (branch != null) {
+                branch.setCommitClosure(new CommitClosure(branch.getRawNode()));
+            }
+            branch = (BitbucketServerBranch) pr.getDestination().getBranch();
+            if (branch != null) {
+                branch.setCommitClosure(new CommitClosure(branch.getRawNode()));
+            }
+        } catch (Exception e) {
+            LOGGER.log( Level.SEVERE, "setupClosureForPRBranch", e);
```

Comment:
The formatting could be improved by removing the trailing space after `LOGGER.log( `.

Reply:
adapted the code